### PR TITLE
[homework #8] - js-advanced-1 by Candelaria Barcelo

### PR DESCRIPTION
**Pull request by candebarcelo**
Hello @VitalikMelnyk ! I am submitting homework 8 + one of the optional tasks. I'm having a little trouble with task number 6, if you could guide me a little I would appreciate it very much. 
Thank you!

### CODE REVIEW

File: homeworks/candelaria-barcelo/08-js-advanced-1/index.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
+// task 1: Pluck function
+const pluck = (object, keyInput) => {
+  try {
+    let path = keyInput.split(".")
+    var current = object;
```
Comment (solvd-mmelnyk) [inline homeworks/candelaria-barcelo/08-js-advanced-1/index.js:5]: why do you use `var` here?

File: homeworks/candelaria-barcelo/08-js-advanced-1/index.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
+// task 1: Pluck function
+const pluck = (object, keyInput) => {
+  try {
+    let path = keyInput.split(".")
```
Comment (solvd-mmelnyk) [inline homeworks/candelaria-barcelo/08-js-advanced-1/index.js:4]: semicolons ; are missed in the end

File: homeworks/candelaria-barcelo/08-js-advanced-1/index.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  for (let i=0; i < arr.length; i++) {
+    let entries = Object.entries(arr[i])
+    for (let i=0; i < entries.length; i++) {
+      answer[entries[i][0]] = entries[i][1]
+    }
+  }
+  return answer
+}
+
+// task 6 https://www.codewars.com/kata/547f1a8d4a437abdf800055c
```
Comment (solvd-mmelnyk) [inline homeworks/candelaria-barcelo/08-js-advanced-1/index.js:97]: are you going to finish this kata 6? or should I merge without this kata?

**Review by solvd-mmelnyk — COMMENTED** [review]
@candebarcelo 
hey! in general really good work!
please fix few things:
1.  make sure that code formatting is correct. I still see semicolons in some cases in some no. please use prettier for it
2. make sure that you are using `let` and `const`  correctly (Here should not be `var`)

let me know if you have any questions

File: homeworks/candelaria-barcelo/08-js-advanced-1/index.js (hunk 1 above)
Comment (candebarcelo) [reply homeworks/candelaria-barcelo/08-js-advanced-1/index.js:5]: Oh I didn't realize that I used var haha, I will change it now.

File: homeworks/candelaria-barcelo/08-js-advanced-1/index.js (hunk 2 above)
Comment (candebarcelo) [reply homeworks/candelaria-barcelo/08-js-advanced-1/index.js]: Corrected in new commit.

File: homeworks/candelaria-barcelo/08-js-advanced-1/index.js (hunk 3 above)
Comment (candebarcelo) [reply homeworks/candelaria-barcelo/08-js-advanced-1/index.js:93]: I will prioritize turning in the homeworks that I haven't turned in yet, and if I have some extra time I will finish this. If you want you can merge it and I'll make a new pull request if I can finish it.

**Comment by candebarcelo** [comment]
I've done all the corrections you said @solvd-mmelnyk , thank you.

**Review by VitalikMelnyk — APPROVED** [review]

**Pull request merged**